### PR TITLE
Python: fix: encrypted_content dropped in reasoning parsing when summaries present

### LINKED ISSUE #4644: Python: [Bug]: OpenAIResponsesClient._parse_response_from_openai drops encrypted_content when reasoning summaries are present
### Description

**What happened?**
When using OpenAIResponsesClient with store=False, reasoning.encrypted_content in include, and summary="auto" (or any non-"none" value), the parsed ChatResponse does not contain encrypted_content in the resulting Content objects' additional_properties. The encrypted content is present in the raw API response but is silently dropped during parsing.

**What did you expect to happen?**
The encrypted_content from the API response should be preserved in additional_properties of the text_reasoning Content object regardless of whether summaries or clear-text reasoning content are also present — matching the round-trip expectation set by _prepare_content_for_openai (line ~1097), which serializes summary, encrypted_content, and id together on a single reasoning item.

**Steps to reproduce issue:**
1. Create an OpenAIResponsesClient and an Agent with InMemoryHistoryProvider
2. Configure options with store=False, reasoning={"effort": "low", "summary": "auto"}, and include=["reasoning.encrypted_content"]
3. Send a prompt to a reasoning model (e.g. gpt-5) using the **non-streaming** path (agent.run() / get_response())
4. Inspect the resulting Content objects of type text_reasoning in the response messages
5. Observe that additional_properties is either None or does not contain "encrypted_content", even though the raw API response includes it

### Code Sample

```markdown
from agent_framework import Agent, AgentSession, InMemoryHistoryProvider
from agent_framework.openai import OpenAIResponsesClient
client = OpenAIResponsesClient(api_key="...")
agent = Agent(
    client=client,
    name="test",
    instructions="You are a helpful assistant.",
    default_options={
        "model": "gpt-5",
        "store": False,
        "reasoning": {"effort": "low", "summary": "auto"},
        "include": ["reasoning.encrypted_content"],
    },
    context_providers=[InMemoryHistoryProvider()],
)
session = AgentSession()
# Non-streaming call
response = await agent.run(session=session, message="What is 2+2?")
# Check reasoning contents
for msg in response.messages:
    for content in msg.contents:
        if content.type == "text_reasoning":
            print(f"text: {content.text!r}")
            print(f"id: {content.id!r}")
            print(f"additional_properties: {content.additional_properties!r}")
            # Expected: additional_properties contains {"encrypted_content": "gAAAA..."}
            # Actual:   additional_properties is None (encrypted_content lost)
```

### Error Messages / Stack Traces

```markdown
No error or exception is raised. The encrypted_content is silently dropped.
```

### Package Versions

agent-framework-core: 1.0.0rc3 openai: 2.24.0

### Python Version

Python 3.13.2

### Additional Context

Root cause is in OpenAIResponsesClient._parse_response_from_openai, in the case "reasoning" branch (~line 1399). Three if blocks use an added_reasoning flag that makes them mutually exclusive:

if item_content (line 1401) — handles clear-text reasoning. Captures summary in additional_properties but not encrypted_content. Sets added_reasoning = True.
if item_summary (line 1415) — handles summary-only. Captures summary text but not encrypted_content. Sets added_reasoning = True.
if not added_reasoning (line 1425) — fallback for encrypted-only reasoning. This is the only branch that captures encrypted_content.
When the API returns a reasoning item with both summary and encrypted_content (the normal response when include=["reasoning.encrypted_content"] and summary != "none"), branch 2 fires and encrypted_content is never extracted.

The serialization side already expects the correct round-trip. _prepare_content_for_openai (~line 1097) constructs a single reasoning item with summary, encrypted_content, and id all present — confirming the intended data model supports all three simultaneously.

**The streaming parser is not affected** due to event ordering: summaries arrive via response.reasoning_summary_text.delta/done events, while encrypted_content is on the response.output_item.done event. The added_reasoning flag is local to the output_item.done handler, so the fallback branch fires and captures encrypted_content. Subsequent content coalescing merges everything correctly.

### PR DESCRIPTION
**Pull request by jgarrison929**
## Motivation and Context

Fixes #4644

When `_parse_response_from_openai` encounters a reasoning item with both `summary` and `encrypted_content`, the `encrypted_content` is silently dropped. This is because the three `if` branches in the `"reasoning"` case use an `added_reasoning` flag that makes only the fallback (no-content, no-summary) branch capture `encrypted_content`.

## Description

This PR:

1. Extracts `encrypted_content` once at the top of the `"reasoning"` case block
2. Propagates it into `additional_properties` in **all three branches** (content, summary-only, fallback) — not just the fallback
3. Adds two regression tests:
   - `test_response_reasoning_preserves_encrypted_content_with_summary` — verifies encrypted_content survives when both content and summary are present
   - `test_response_reasoning_preserves_encrypted_content_summary_only` — verifies encrypted_content survives when only summary (no clear-text content) is present

The streaming parser is **not affected** (event ordering means encrypted_content is captured correctly via the `response.output_item.done` handler).

The serialization side (`_prepare_content_for_openai`) already expects `encrypted_content` in `additional_properties`, so this fix restores correct round-trip behavior.

## Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR title and description match the bug report
- [x] Tests have been added that prove the fix works
- [x] All existing unit tests pass (135 passed, integration tests skipped — they require API keys)
- [x] The change is minimal and focused on the specific bug

### CODE REVIEW

File: python/packages/openai/tests/openai/test_openai_chat_client.py [hunk 1]
```diff
@@ -528,6 +528,83 @@ def test_response_content_creation_with_reasoning() -> None:
...
+    mock_response.output = [mock_reasoning_item]
+
+    response = client._parse_response_from_openai(mock_response, options={})  # type: ignore
+
+    # The content branch should carry encrypted_content in additional_properties
+    reasoning_contents = [c for c in response.messages[0].contents if c.type == "text_reasoning"]
+    assert len(reasoning_contents) >= 1
+    first_reasoning = reasoning_contents[0]
+    assert first_reasoning.text == "Reasoning step"
+    assert first_reasoning.additional_properties is not None
```
Comment (moonbox3) [inline python/packages/openai/tests/openai/test_openai_chat_client.py:574]: This test only asserts on `reasoning_contents[0]` (the content branch). The code change also adds `encrypted_content` to the summary branch. The second reasoning element should also be checked:
```python
assert reasoning_contents[1].additional_properties is not None
assert reasoning_contents[1].additional_properties.get('encrypted_content') == 'gAAAA_encrypted_payload'
```

**Review by moonbox3 — COMMENTED** [review]
## Automated Code Review

**Reviewers:** 4 | **Confidence:** 90%

### ✓ Correctness
> The diff correctly fixes a bug (#4644) where `encrypted_content` was silently dropped when a reasoning item contained both summaries/content and encrypted payload. The fix extracts `encrypted_content` once at the top of the reasoning case block and propagates it through all three branches (content, summary-only, fallback). The two new regression tests properly cover the content+summary+encrypted and summary-only+encrypted scenarios. The logic, data flow, and test assertions are all correct.

### ✗ Security Reliability
> The fix correctly propagates `encrypted_content` through all branches of the non-streaming `_parse_response_from_openai` method, preventing silent data loss. However, the parallel streaming code path (`_process_streaming_events`, lines ~2036-2070) contains the exact same bug pattern — `encrypted_content` is only captured in the `not added_reasoning` fallback branch and is dropped when `content` is present — and is left unfixed. This is a reliability concern because encrypted reasoning payloads will still be silently lost during streaming responses. No security issues were found with the change itself.

### ✗ Test Coverage
> The two new tests properly cover the main regression from #4644 (content+summary+encrypted and summary-only+encrypted paths). However, there are gaps: (1) the pre-existing `test_response_content_creation_with_reasoning` test uses a bare MagicMock for `mock_reasoning_item` without setting `encrypted_content = None`, meaning `getattr(item, 'encrypted_content', None)` now returns a truthy MagicMock (since MagicMock auto-creates attributes), silently injecting a bogus encrypted_content into additional_properties — this test should be updated; (2) there is no dedicated test for the refactored fallback branch (no content, no summary, only encrypted_content) which was the only branch that previously worked correctly; (3) there is no negative test asserting encrypted_content is absent from additional_properties when the field is None/missing.

### ✗ Design Approach
> The fix correctly addresses the reported bug: `encrypted_content` was silently dropped on the non-streaming parse path when `content` or `summary` items were also present. The implementation — extracting `encrypted_content` once and propagating it into every branch — is sound for the common single-step case. Two issues are worth noting: (1) The identical streaming parse path (around line 2060) has the exact same bug but was not patched, making this an overly narrow fix. (2) When a reasoning item has N content steps, `encrypted_content` is stamped onto each of the N Content objects; on round-trip, `_prepare_content_for_openai` emits each Content as a separate `{"type": "reasoning", "encrypted_content": ...}` API object, duplicating the payload N times. The multi-step duplication is a pre-existing structural problem (splitting one reasoning item into N Content objects), but attaching `encrypted_content` to every sub-item worsens it.

### Flagged Issues
- [ ] The streaming path in `_process_streaming_events` (~lines 2036-2070 of `_responses_client.py`) has the identical `encrypted_content` data-loss bug: `encrypted_content` is only captured in the `not added_reasoning` fallback branch and is dropped whenever `event_item.content` is present. The fix is incomplete—streaming callers will still silently lose encrypted payloads. The same extraction-and-propagation pattern from the non-streaming fix should be applied here.
- [ ] The existing `test_response_content_creation_with_reasoning` (line 513) creates `mock_reasoning_item = MagicMock()` without setting `encrypted_content = None`. Because MagicMock auto-creates attributes, `getattr(item, 'encrypted_content', None)` now returns a truthy MagicMock, silently injecting a fake `encrypted_content` into every reasoning Content's `additional_properties`. The mock should explicitly set `mock_reasoning_item.encrypted_content = None`.

### Suggestions
- Apply the same `encrypted_content = getattr(event_item, 'encrypted_content', None)` extraction at the top of the `case 'reasoning'` block in the streaming path (~line 2037), and propagate it into `additional_properties` in both the content-iteration loop and the summary/fallback branches, mirroring the non-streaming fix.
- Add a test for the fallback branch (no content, no summary, only `encrypted_content`) to ensure the refactoring from inline `getattr` to the pre-extracted variable didn't break the original working path.
- In `test_response_reasoning_preserves_encrypted_content_with_summary`, also assert that the summary-branch Content (`reasoning_contents[1]`) carries `encrypted_content`, not just the first content-branch element.
- Add a negative test where `encrypted_content` is None, confirming `additional_properties` is either None or does not contain the `encrypted_content` key, to guard against accidental truthy leaks.
- When `item.content` has multiple entries, attaching `encrypted_content` to every Content object causes the round-trip through `_prepare_content_for_openai` to emit duplicate `{"type": "reasoning", "encrypted_content": ...}` API objects. Consider attaching it only to the first item (index == 0) to avoid payload duplication.

---
*Automated review by moonbox3's agents*

File: python/packages/openai/tests/openai/test_openai_chat_client.py (hunk 1 above)
Comment (jgarrison929) [reply python/packages/openai/tests/openai/test_openai_chat_client.py:574]: Addressed in ff01ba3. Added assertion on reasoning_contents[1] for the summary branch. Also fixed the streaming path (same encrypted_content bug), fixed MagicMock leak in existing tests, and added a negative test for the None case.

**Comment by jgarrison929** [comment]
Thanks for the thorough review @moonbox3. All items addressed in ff01ba3:

- **Streaming path fixed**: Applied same `encrypted_content` extraction pattern to `_process_streaming_events` case 'reasoning' block
- **MagicMock leak fixed**: Set `encrypted_content = None` on existing reasoning test mocks to prevent truthy auto-attribute injection
- **Summary branch assertion added**: `reasoning_contents[1]` now verified to carry `encrypted_content`
- **Negative test added**: `test_response_reasoning_no_encrypted_content` confirms clean `additional_properties` when `encrypted_content` is None

136/136 tests pass (1 pre-existing integration skip requiring API key).

**Comment by moonbox3** [comment]
There are a couple of conflicts, @jgarrison929. Could you have a look, please?

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/README.md">Python Test Coverage</a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>packages/openai/agent_framework_openai</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py">_chat_client.py</a></td><td>896</td><td>133</td><td>85%</td><td><a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L346">346</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L348-L349">348&ndash;349</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L406-L407">406&ndash;407</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L409">409</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L441">441</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L500-L503">500&ndash;503</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L507-L508">507&ndash;508</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L513-L514">513&ndash;514</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L524-L525">524&ndash;525</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L532">532</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L547-L553">547&ndash;553</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L574">574</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L582">582</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L605">605</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L702">702</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L801">801</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L860">860</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L862">862</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L864">864</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L866">866</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L932">932</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L946">946</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1026">1026</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1036">1036</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1041">1041</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1084">1084</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1163">1163</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1180">1180</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1193">1193</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1258">1258</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1351">1351</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1356">1356</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1360-L1362">1360&ndash;1362</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1366-L1367">1366&ndash;1367</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1433">1433</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1462">1462</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1468">1468</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1478">1478</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1484">1484</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1489">1489</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1495">1495</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1500-L1501">1500&ndash;1501</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1562">1562</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1584-L1585">1584&ndash;1585</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1600-L1601">1600&ndash;1601</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1619-L1620">1619&ndash;1620</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1674-L1677">1674&ndash;1677</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1839">1839</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1894">1894</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1896">1896</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L1976-L1984">1976&ndash;1984</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2106">2106</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2141">2141</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2156">2156</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2176-L2186">2176&ndash;2186</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2199">2199</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2210-L2214">2210&ndash;2214</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2228">2228</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2242-L2253">2242&ndash;2253</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2262">2262</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2301-L2304">2301&ndash;2304</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2312-L2313">2312&ndash;2313</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2315-L2317">2315&ndash;2317</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2331-L2333">2331&ndash;2333</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2343-L2344">2343&ndash;2344</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2350">2350</a>, <a href="https://github.com/microsoft/agent-framework/blob/49a56cb9f688081978ef4425328e9a261d0c2483/packages/openai/agent_framework_openai/_chat_client.py#L2365">2365</a></td></tr><tr><td><b>TOTAL</b></td><td><b>28021</b></td><td><b>3412</b></td><td><b>87%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 5457 | 20 :zzz: | 0 :x: | 0 :fire: | 1m 22s :stopwatch: |

**Comment by jgarrison929** [comment]
Rebased against main and resolved conflicts. Should be clean now. Thanks for the heads up @moonbox3!

**Comment by jgarrison929** [comment]
Hi @moonbox3 — conflicts are resolved and rebased. The dotnet CI failure is an unrelated flaky test in `MessageMergerTests.Test_MessageMerger_AssemblesMessage` (timestamp race condition — "Did not expect the date and time to represent the same point in time"). This PR only touches Python files. Looks like a known issue given the ~20 other Workflows tests already marked as "Flaky test - temporarily disabled." Happy to re-trigger if needed!

**Comment by moonbox3** [comment]
Please make sure you've got the `prek` hooks installed so that the code quality issues are caught before merging and pushing. See [here](https://github.com/microsoft/agent-framework/blob/main/python/DEV_SETUP.md#after-installing-uv) for more info.

**Comment by jgarrison929** [comment]
Ran `poe check` (prek hooks) and pushed fixes — replaced an en-dash with a hyphen in a comment (RUF003) and updated the test file to use the renamed `OpenAIChatClient` class + `model=` parameter (F821). All 50 lint/fmt tasks, 25 pyright tasks, and openai package tests pass clean now. Thanks for the pointer @moonbox3! 🙏

**Review by moonbox3 — APPROVED** [review]

**Comment by eavanvalkenburg** [comment]
Not sure if this is still relevant, there have been some updates to the parsing of encrypted reasoning content. Closing as part of cleanup of PR backlog, feel free to reopen if still relevant.
